scripts/sync_detection_rules.py

```diff
@@ -0,0 +1,866 @@
+
+    Args:
+        pr_sha (str): SHA of the PR head commit
+        action_name (str): Name of the action/check to look for
+        required_status (str): Required status (success, failure, etc.)
+
+    Returns:
+        bool: True if the action has completed with the required status, False otherwise
+    """
+    # Use the GitHub Checks API to get all check runs for this commit
+    url = f'https://api.github.com/repos/{REPO_OWNER}/{REPO_NAME}/commits/{pr_sha}/check-runs'
+    custom_headers = headers.copy()
+    # Add the required Accept header for the Checks API
+    custom_headers['Accept'] = 'application/vnd.github.v3+json'
+
+    response = requests.get(url, headers=custom_headers)
+
+    if response.status_code != 200:
+        print(f"\tError checking action status: {response.status_code}")
+        return False
+
+    check_runs = response.json()
+
+    if 'check_runs' not in check_runs or len(check_runs['check_runs']) == 0:
+        print(f"\tNo check runs found for commit {pr_sha}")
+        return False
+
+    # Look for the specific action by name
+    for check in check_runs['check_runs']:
+        check_name = check['name']
+        check_conclusion = check['conclusion']
+        check_status = check['status']
+
+        if action_name.lower() in check_name.lower():
+
+            # Check if the action is complete
+            if check_status != 'completed':
+                print(f"\tCheck '{check_name}' is still in progress (status: {check_status})")
+                return False
+
+            # Check if the action has the required conclusion
+            if check_conclusion == required_status:
+                return True
+            else:
+                print(f"\tCheck '{check_name}' has conclusion '{check_conclusion}', expected '{required_status}'")
+                return False
+
+    print(f"\tNo check matching '{action_name}' found")
+    return False
+
+def contains_skip_text(content, skip_text):
+    """
+    Check if file content contains the text to skip.
+
+    Args:
+        content (str): File content
+        skip_text (str): Text to search for
+
+    Returns:
+        bool: True if content contains the skip text, False otherwise
+    """
+    return skip_text in content
+
+
+def generate_deterministic_uuid(seed_string):
+    """
+    Generate a deterministic UUID based on a seed string.
+    This ensures the same input will always produce the same UUID.
+
+    Args:
+        seed_string (str): A string to use as a seed for UUID generation
+
+    Returns:
+        str: A UUID string in the format of XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    """
+    # Create a namespace UUID (using the DNS namespace as a standard practice)
+    namespace = uuid.UUID('6ba7b810-9dad-11d1-80b4-00c04fd430c8')
+
+    # Create a UUID using the namespace and the seed string
+    deterministic_uuid = uuid.uuid5(namespace, seed_string)
+
+    return str(deterministic_uuid)
+
+
+def add_id_to_yaml(content, filename):
```

Comment:
I'm curious why not just load the yaml using `pyyaml` and then update and write the new yaml file back to disk instead of using regex? I'm sure you have a reason just wondering.

Reply:
pyyaml doesn't output the source in the same format and it becomes very messy